### PR TITLE
Improvements to detect & handle Beta versions.

### PR DESCRIPTION
**Pull request by Martinski4GitHub**
Code improvements to correctly detect & handle Beta (and Alpha) version string formats.
Cleaning up some code.

**NOTE**:
For the record, our current position is still that "Alpha-to-Production" updates are not **officially** supported.
We're making only a best-effort attempt to detect Alpha versions and update to the Master version.

### CODE REVIEW

**Comment by ExtremeFiretop** [comment]
> Code improvements to correctly detect & handle Beta (and Alpha) version string formats. Cleaning up some code.
> 
> **NOTE**: For the record, our current position is still that "Alpha-to-Production" updates are not **officially** supported. We're making only a best-effort attempt to detect Alpha versions and update to the Master version.

Just waking up and starting my day now. My eyes are stuff blurry but I'll get to reviewing this momentarily!

Thanks as always! for not just your input, but your knowledge and understanding of the issues and how to address them. It's a beautiful thing to see in action.

File: MerlinAU.sh [hunk 1]
```diff
@@ -1728,7 +1732,7 @@ _GetAvailableRAM_KB_()
    # Since not all Page Cache is guaranteed to be reclaimed at any
    # moment, we simply estimate that only half will be reclaimable.
    #----------------------------------------------------------------#
-   theMemAvailable_KB="$((theMemFree_KB + ((thePageCache_KB / 2))))"
+   theMemAvailable_KB="$((theMemFree_KB + (thePageCache_KB / 2)))"
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:1735]: Approved

File: MerlinAU.sh [hunk 2]
```diff
@@ -1668,24 +1674,22 @@ _GetCurrentFWInstalledLongVersion_()
...
-## Modified by Martinski W. [2023-Nov-22] ##
+## Modified by Martinski W. [2024-Feb-28] ##
 ##----------------------------------------##
 _GetCurrentFWInstalledShortVersion_()
 {
-##FOR DEBUG ONLY##
-if true
-then
+##FOR TESTING/DEBUG ONLY##
+if false ; then echo "388.5.0" ; return 0 ; fi
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:1682]: Approved

File: MerlinAU.sh [hunk 3]
```diff
@@ -1668,24 +1674,22 @@ _GetCurrentFWInstalledLongVersion_()
...
+##FOR TESTING/DEBUG ONLY##
+
     local theVersionStr  extVersNum
 
     extVersNum="$(nvram get extendno | awk -F '-' '{print $1}')"
+    echo "$extVersNum" | grep -qiE "^(alpha|beta)" && extVersNum="0_$extVersNum"
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:1688]: Approved

File: MerlinAU.sh [hunk 4]
```diff
@@ -1649,16 +1654,17 @@ _AddPostRebootRunScriptHook_()
...
+## Modified by Martinski W. [2024-Feb-28] ##
+##----------------------------------------##
 _GetCurrentFWInstalledLongVersion_()
 {
    local theBranchVers  theVersionStr  extVersNum
 
    theBranchVers="$(nvram get firmver | sed 's/\.//g')"
 
    extVersNum="$(nvram get extendno)"
+   echo "$extVersNum" | grep -qiE "^(alpha|beta)" && extVersNum="0_$extVersNum"
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:1667]: Approved

File: MerlinAU.sh [hunk 5]
```diff
@@ -383,14 +383,19 @@ _FWVersionStrToNum_()
...
+    #--------------------------------------------------------------
+    if echo "$verStr" | grep -qiE '(alpha|beta)'
     then
-        nonProductionVersionWeight=-100
-        # Remove 'alpha/beta' and any following numbers #
-        verStr="$(echo "$verStr" | sed 's/[._-]\?[Aa]lpha[0-9]*// ; s/[._-]\?[Bb]eta[0-9]*//')"
+        # Adjust weight value if "Beta-to-Production" update is enabled #
+        [ "$USE_BETA_WEIGHT" = "ENABLED" ] && nonProductionVersionWeight=-100
+
+        # Replace '.alpha|.beta' and anything following it with ".0" #
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:395]: Approved

File: MerlinAU.sh [hunk 6]
```diff
@@ -383,14 +383,19 @@ _FWVersionStrToNum_()
...
+        verStr="$(echo "$verStr" | sed 's/[.][Aa]lpha.*/.0/ ; s/[.][Bb]eta.*/.0/')"
+        # Remove 'alpha|beta' and anything following it #
+        verStr="$(echo "$verStr" | sed 's/[_-]\?[Aa]lpha.*// ; s/[_-]\?[Bb]eta.*//')"
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:398]: Approved

File: MerlinAU.sh [hunk 7]
```diff
@@ -1226,12 +1231,12 @@ _CreateEMailContent_()
...
+   subjectStr="F/W Update Status for $MODEL_ID"
    fwInstalledVersion="$(_GetCurrentFWInstalledLongVersion_)"
-   case "$fwInstalledVersion" in
-      *_rog) fwInstalledVersion=$(echo "$fwInstalledVersion" | sed 's/_rog$//') ;;
-   esac
    fwNewUpdateVersion="$(_GetLatestFWUpdateVersionFromRouter_ 1)"
-   subjectStr="F/W Update Status for $MODEL_ID"
+
+   # Remove "_rog" suffix to avoid version comparison failures #
+   fwInstalledVersion="$(echo "$fwInstalledVersion" | sed 's/_rog$//')"
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:1239]: Approved

**Review by ExtremeFiretop — APPROVED** [review]

**Comment by ExtremeFiretop** [comment]
Tested with success. Merging.

**Comment by ExtremeFiretop** [comment]
> > Just waking up and starting my day now. My eyes are still blurry but I'll get to reviewing this momentarily!
> > Thanks as always! for not just your input, but your knowledge and understanding of the issues and how to address them. It's a beautiful thing to see in action.
> 
> Just waking up myself too... I've got an early morning meeting today in about an hour & 15 minutes. I'm working from home today and have to update some functional specs as well.
> 
> Anyway, reading the latest forum posts, found someone reporting an error WRT the HSTS database when our script is running. I know what the problem is and it's an easy fix. I'll submit a PR in about 10 minutes.

It's not even an actual bug or problem to the script, it works how it's intended, but I can see how this can be confusing to users. I didn't see it being a critical thing to fix but if you want to address it I'm more than willing to test and review because I also often see the HSTS warning.

**Comment by Martinski4GitHub** [comment]
> 
> It's not even an actual bug or problem to the script, it works how it's intended, but I can see how this can be confusing to users. I didn't see it being a critical thing to fix but if you want to address it I'm more than willing to test and review because I also often see the HSTS warning.
>
Oh yeah, it's not a fatal error; but it's displayed as an ERROR, not a warning message, like this:

"ERROR: could not open HSTS store at '/root/.wget-hsts' HSTS will be disabled."

So I can see some users getting concerned that something went wrong and reporting it as a problem. Since it was a very easy & quick fix, I decided to just do it. One less thing that users will report as an "error" by the script.

**Pull request merged**